TicketBookingSystemNew.sol

```diff
@@ -0,0 +1,357 @@
+            seatNumber: _seatNumber,
+            mintedBy: _mintedBy
+        });
+
+        uint256 newTokenId = tokenId;
+        _safeMint(receiver, newTokenId);
+        tokenId += 1;
+        return newTokenId;
+    }
+
+    //helper function to get the address who minted the ticket
+    function mintedBy(uint256 _tokenId) external view returns (address) {
+        return ticketInfoMapping[_tokenId].mintedBy;
+    }
+
+    //helper function to get show data from ticket id
+    function getShowDateFromTicketId(uint256 _tokenId)
+        external
+        view
+        returns (uint256)
+    {
+        return ticketInfoMapping[_tokenId].showDate;
+    }
+
+    //check if ticket exists
+    function exists(uint256 _tokenId) public view returns (bool) {
+        return _exists(_tokenId);
+    }
+
+    //burn the ticket
+    function burn(uint256 tokenId) public {
+        _burn(tokenId);
+    }
+}
+
+contract Poster is ERC721, Ownable {
+    uint256 private posterTokenId;
+
+    constructor() ERC721("Poster", "PSTR") {
+        uint256 posterTokenId = 0;
+    }
+
+    //mint new poster
+    function mintNewPoster(address _to) external onlyOwner returns (uint256) {
+        uint256 newTokenId = posterTokenId;
+        _safeMint(_to, newTokenId);
+        posterTokenId += 1;
+        return posterTokenId;
+    }
+}
+
+contract TicketBookingSystem {
+    address payable public salesManager;
+    address public ticketBuyer;
+    string constant _seatViewLink = "https://seatplan.com/";
+    uint256 numberOfSeatsPerRow = 10;
+    uint256 numberOfRows = 2;
+    uint256 totalNumberOfSeats = numberOfSeatsPerRow * numberOfRows;
+    uint256 availableTickets = totalNumberOfSeats;
+    uint256 showDate;
+    string showTitle;
+    uint256 price;
+    ShowStatus showStatus;
+    Ticket ticketInstance;
+    Poster posterInstance;
+
+    //this mapping stores amount to be refunded against each address if show gets cancelled
+    mapping(address => uint256) amountToBeRefunded;
+
+    //this mapping stores a Seat struct against each seat index (1-60)
+    mapping(uint256 => Seat) public seats;
+
+    //this mapping stores an array of ticketIds bought by each address
+    //each address can purchase multiple tickets
+    mapping(address => uint256[]) ticketsSold;
+
+    enum ShowStatus {
+        Scheduled,
+        Over,
+        Cancelled
+    }
+
+    constructor(
+        string memory _showTitle,
+        uint256 _showDate,
+        uint256 _ticketPrice
+    ) payable {
+        bytes memory tempEmptyStringTest = bytes(_showTitle);
+
+        //check if show title is empty string
+        require(tempEmptyStringTest.length != 0, "Show title can not be empty");
+
+        //check if show date is in past
+        require(
+            _showDate >= block.timestamp,
+            "Show can not be scheduled in past"
+        );
+        //check if ticket price is > 0
+        require(_ticketPrice > 0, "Ticket price must be greater than 0");
```

Comment:
Is this necessary since the input is uint?

Reply:
hmm, valid point. maybe we can remove this check?